tests/python_client/testcases/test_query.py

```diff
@@ -1190,3 +1195,88 @@ def test_query_growing_segment_data(self):
+        res = vectors[0].iloc[:2, :3].to_dict('records')
+        output_fields = [default_float_field_name, default_string_field_name]
+        collection_w.query(default_string_term_expr, output_fields=output_fields, 
+                           check_task=CheckTasks.check_query_results, check_items={exp_res: res})
+    
+    @pytest.mark.tags(CaseLabel.L1)
+    @pytest.mark.parametrize("expression", cf.gen_normal_string_expressions(default_string_field_name))
+    def test_query_string_is_primary(self, expression):
+        """
+        target: test query with output field only primary field
+        method: specify string primary field as output field
+        expected: return string primary field
+        """
+        collection_w, vectors = self.init_collection_general(prefix, insert_data=True, primary_field=ct.default_string_field_name)[0:2]
+        res, _ = collection_w.query(expression, output_fields=[ct.default_string_field_name])
+        assert list(res[0].keys()) == [ct.default_string_field_name]
+
+
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_query_string_with_mix_expr(self):
+        """
+        target: test query data
+        method: create collection and insert data
+                collection.load()
+                query with mix expr in string field and int field
+        expected: query successfully
+        """
+        collection_w, vectors = self.init_collection_general(prefix, insert_data=True, primary_field=ct.default_string_field_name)[0:2]
+        collection_w.load()
+        res = vectors[0].iloc[:, 1:3].to_dict('records')
+        output_fields = [default_float_field_name, default_string_field_name]
+        collection_w.query(default_mix_expr, output_fields=output_fields, 
+                                   check_task=CheckTasks.check_query_results, check_items={exp_res: res})
+
+
+    @pytest.mark.tags(CaseLabel.L1)
+    @pytest.mark.parametrize("expression", cf.gen_invaild_string_expressions())
+    def test_query_with_invalid_string_expr(self, expression):
+        """
+        target: test query data
+        method: create collection and insert data
+                collection.load()
+                query with invalid expr
+        expected: Raise exception
+        """
+        collection_w = self.init_collection_general(prefix, insert_data=True)[0]
+        collection_w.load()
+        collection_w.query(expression, check_task=CheckTasks.err_res,
+                           check_items={ct.err_code: 1, ct.err_msg: "type mismatch"})
+       
+
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_query_string_expr_with_binary(self):
+        """
+        target: test query string expr with binary
+        method: query string expr with binary 
+        expected: verify query successfully
+        """
+        collection_w, vectors= self.init_collection_general(prefix, insert_data=True, is_binary=True)[0:2]
+        collection_w.create_index(ct.default_binary_vec_field_name, binary_index_params)
```

Comment:
After create_index, load() should be added if you want index you created is valid before query.